sherlock-platform/customization/resources/META-INF/SherlockPlatformCustomization.xml

```diff
@@ -0,0 +1,14 @@
+<!-- Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+<idea-plugin>
+
+    <depends>com.intellij.modules.platform</depends>
+    <actions resource-bundle="messages.ActionsBundle">
+        <group id="WelcomeScreen.Platform.NewProject">
+            <action id="NewProjectStep"
+                    class="com.google.sherlock.onboarding.NewProjectStep"
+                    text="New Project"/>
+            <reference ref="WelcomeScreen.OpenDirectoryProject"/>
```

Comment:
When I hit NewProject, or when I open an existing project (that I created using vanilla IDEA), I get this exception thrown:

```
2024-11-07 10:53:48,411 [   8000] SEVERE - #c.i.f.ProductivityFeaturesRegistryImpl - ProductivityFeaturesRegistry.xml not found
java.lang.Throwable: ProductivityFeaturesRegistry.xml not found
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:376)
	at com.intellij.featureStatistics.ProductivityFeaturesRegistryImpl.reloadFromXml(ProductivityFeaturesRegistryImpl.java:69)
	at com.intellij.featureStatistics.ProductivityFeaturesRegistryImpl.<init>(ProductivityFeaturesRegistryImpl.java:49)
	at com.intellij.platform.instanceContainer.instantiation.InstantiateKt$instantiate$2.invoke(instantiate.kt:45)
...
```

Looking across IntelliJ/Cidr/Pycharm, I see that we can fix this exception by adding a `ProductivityFeaturesRegistry.xml`  file under `sherlock-platform/customization/resources/SherlockPlatformCustomization.xml` . It's contents can be:

```
<component>
</component>
```

Reply:
This is weird as I hit no such error. When I click on `New Project` nothing happens and when I click on `Open Project` I am able to any existing project.

Reply:
I was still able to `Open Project` and click on `New Project` without any errors. @emrekultursay can you please try after clearing yours?

Reply:
It's not an error, but I can see the exceptions being thrown in the background.

Reply:
I found why this behaves differently when we build from IDEA vs. using `build_sherlock_platform.sh`:

When we use `build_sherlock_platform.sh`, our Properties file extends `BaseIdeaProperties`, which contains [this line]( https://github.com/android-graphics/sherlock-platform/blob/3ed3b70e7ccf09e1bb858436e625cf9af3748c0b/platform/build-scripts/src/org/jetbrains/intellij/build/BaseIdeaProperties.kt#L124):

```
layout.withModule("intellij.java.ide.resources")
```

Whereas when we build from IDEA, we don't have a dependency on that module. The missing file  `ProductivityFeaturesRegistry.xml` comes from that module. 

So, we can:
  1. Comment that line out, add our own empty `ProductivityFeaturesRegistry.xml` (similar to PyCharm)
  2. Add a dependency to `intellij.java.ide.resources` to our `intellij.idea.community.main.sherlock.iml` (we will probably need to revisit this later)
I verified that both of them work.

Reply:
Ahh, makes sense now. I had only tried through `build_sherlock_platform.sh`. I agree, adding a dependency should be sufficient for now. Will update.

Reply:
Updated, PTAL